src/crypto/public_key/elliptic/ecdh_x.js

```diff
@@ -24,12 +25,27 @@ const HKDF_INFO = {
  */
 export async function generate(algo) {
   switch (algo) {
-    case enums.publicKey.x25519: {
-      // k stays in little-endian, unlike legacy ECDH over curve25519
-      const k = getRandomBytes(32);
-      const { publicKey: A } = x25519.box.keyPair.fromSecretKey(k);
-      return { A, k };
-    }
+    case enums.publicKey.x25519:
+      try {
+        const webCrypto = util.getWebCrypto();
+        // Note: keys generated with ECDSA and ECDH are structurally equivalent
+        const webCryptoKey = await webCrypto.generateKey({ name: 'X25519' }, true, ['deriveKey', 'deriveBits']);
```

Comment:
You can also pass the name directly:
```suggestion
        const webCryptoKey = await webCrypto.generateKey('X25519', true, ['deriveKey', 'deriveBits']);
```